src/ophyd_async/sim/demo/sim_motor.py

```diff
@@ -54,7 +54,7 @@ def set(self, new_position: float, timeout: Optional[float] = None) -> AsyncStat
         """
         watchers: List[Callable] = []
         coro = asyncio.wait_for(self._move(new_position, watchers), timeout=timeout)
-        return AsyncStatus(coro, watchers)
+        return AsyncStatus(coro)
 
     async def _move(self, new_position: float, watchers: List[Callable] = []):
```

Comment:
Please can we use the same `wrap` logic here? Something like

```suggestion
    @WatchableAsyncStatus.wrap
    async def set(self, new_position: float) -> :
```

then yielding `WatcherUpdate` instead of calling watchers